### ISSUE TITLE
ValueError: Object arrays cannot be loaded when allow_pickle=False in `recv()`

### ISSUE THREAD

**Issue by amanning32** [sabjorn/NumpySocket#8]
### Description

I'm using the newest version (1.0.0). When trying to use numpysocket with numpy >= 1.17.0, there is an error in recv(). Stack trace (omitted calling code from my function, which just performs `npsocket.recv()`, as well as parts of the path):
```
  File "...\site-packages\numpysocket\numpysocket.py", line 45, in recv
    frame = np.load(BytesIO(frameBuffer))['frame']
  File "...\site-packages\numpy\lib\npyio.py", line 245, in __getitem__
    return format.read_array(bytes,
  File "...\site-packages\numpy\lib\format.py", line 741, in read_array
    raise ValueError("Object arrays cannot be loaded when "
ValueError: Object arrays cannot be loaded when allow_pickle=False
```

### To Reproduce
I didn't run these commands, but this should reproduce the error (it's a simplified example of our use case):
- Use `numpy>=1.17.0`
- Create an `np.array` with `dtype=obj`, like `a = np.array([[1], [1, 2]], dtype=obj)` (Note: omitting this in numpy>=1.19.0 gives a VisibleDeprecationWarning: https://stackoverflow.com/questions/63097829/debugging-numpy-visibledeprecationwarning-ndarray-from-ragged-nested-sequences)
- Send the array: `npsocket.sendall(a)`
- Try to receive the array: `npsocket.recv()`. This will provide the error above.

### Tasks
In numpy==1.17.0, the default of `np.load(...allow_pickle=True...)` was changed from True to False. Explicitly setting True should fix the problem (see https://stackoverflow.com/questions/55824625/how-to-fix-object-arrays-cannot-be-loaded-when-allow-pickle-false-in-the-sketc?answertab=votes#tab-top)

- Change numpysocket/numpysocket.py:45 to ```frame = np.load(BytesIO(frameBuffer), allow_pickle=True)['frame']```

I did this locally (on my install) and was able to resolve the error.

This is my first GitHub issue, so please let me know if there is any other information I can provide to help!

**Comment by sabjorn** [comment]
@amanning32 thank you for the writeup. I greatly appreciate especially since it already has the solution for me. This also really shows how important tests (which I have put off for far too long :| ). I'll let you know when I get this done, should be pretty soon.

Also, unrelated, I would be very interested to know what you use this library for. I have a vague notion that there are a number of people out there using it BUT I haven't developed much of a community around this project.

**Comment by sabjorn** [comment]
@amanning32 can you take a look at this PR and see if this fix works as expected?

https://github.com/sabjorn/NumpySocket/pull/10

**Comment by amanning32** [comment]
@sabjorn Yep, that PR is exactly what I had done to get it working locally, so I think that should work as expected!

And for the use case, I'm (currently) a Master's student at Georgia Tech; I'm taking a class on Systems for Machine Learning, and for our class research project, we're looking at compressing updates for machine learning models deployed at the edge. (We're basically taking this paper and extending it to RNNs and LSTMs: https://arxiv.org/abs/2203.04516)

I don't really have a machine learning background, so I was tasked with building the pipeline/system; this sort of approach was the first one I thought of for how to send an update package from a central server to another device, so I'm glad this project already existed--it would've taken me much longer to figure out how to do it myself!

**Comment by sabjorn** [comment]
fixed with 6438c88ec7b5d7a6dd93326e5b04e3bd7af525a9

**Comment by sabjorn** [comment]
@amanning32 thanks for the info about the project. I am very happy to hear that this library is being used (and saved you a lot of time). 

I hope the update I pushed worked for you (v1.0.1). If you have any other issued, don't hesitate to post an issue. And good luck with the project.
